packages/libs/coreui/src/components/inputs/SingleSelect.tsx

```diff
@@ -1,142 +1,156 @@
+   * pointing to the same object reference as what's used in the `items` prop
+   */
+  value: T;
 
-    onSelect: (value: T) => void;
-    buttonDisplayContent: ReactNode;
-    isDisabled?: boolean;
+  onSelect: (value: T) => void;
+  buttonDisplayContent: ReactNode;
+  isDisabled?: boolean;
 }
 
+const checkIconContainer = { height: 16, width: 16 };
+
 export default function SingleSelect<T>({
-    items,
-    value,
-    onSelect,
-    buttonDisplayContent,
-    isDisabled = false,
+  items,
+  value,
+  onSelect,
+  buttonDisplayContent,
+  isDisabled = false,
 }: SingleSelectProps<T>) {
-    const [ isPopoverOpen, setIsPopoverOpen ] = useState<boolean>(false);
+  const [isPopoverOpen, setIsPopoverOpen] = useState<boolean>(false);
 
-    /** 
-     * 1. Find the index of the value prop in the items array to set focused state in the dropdown 
-     * 2. If a value is not found, defaults to the first item in the dropdown
-    */
-    const selectedValueIndex = items.findIndex(item => value === item.value);
-    const defaultOrSelectedValueIndex = selectedValueIndex !== -1 ? selectedValueIndex : 0;
-    const [ indexOfFocusedElement, setIndexOfFocusedElement ] = useState(defaultOrSelectedValueIndex);
+  /**
+   * 1. Find the index of the value prop in the items array to set focused state in the dropdown
+   * 2. If a value is not found, defaults to the first item in the dropdown
+   */
+  const selectedValueIndex = items.findIndex((item) => value === item.value);
+  const defaultOrSelectedValueIndex =
+    selectedValueIndex !== -1 ? selectedValueIndex : 0;
+  const [indexOfFocusedElement, setIndexOfFocusedElement] = useState(
+    defaultOrSelectedValueIndex
+  );
 
-    const [ key, setKey ] = useState<string>('');
+  const [key, setKey] = useState<string>('');
 
-    const handleSelection = (newValue: T) => {
-        onSelect(newValue);
-        setKey(uniqueId());
-    }
+  const handleSelection = (newValue: T) => {
+    onSelect(newValue);
+    setKey(uniqueId());
+  };
 
-    /** Update focused element when index of selected value changes */
-    useEffect(() => {
-        setIndexOfFocusedElement(defaultOrSelectedValueIndex)
-    }, [defaultOrSelectedValueIndex])
+  /** Update focused element when index of selected value changes */
+  useEffect(() => {
+    setIndexOfFocusedElement(defaultOrSelectedValueIndex);
+  }, [defaultOrSelectedValueIndex]);
 
-    const onKeyDown = (key: string, newValue: T) => {
-        if (!isPopoverOpen) return;
-        if (key === 'Enter') {
-            handleSelection(newValue)
-        } else if (key === 'ArrowUp') {
-            indexOfFocusedElement !== 0 ? setIndexOfFocusedElement(prev => prev - 1) : null;
-        } else if (key === 'ArrowDown') {
-            indexOfFocusedElement !== items.length - 1 ? setIndexOfFocusedElement(prev => prev + 1) : null;
-        }
+  const onKeyDown = (key: string, newValue: T) => {
+    if (!isPopoverOpen) return;
+    if (key === 'Enter') {
```

Comment:
**suggestion**: If we wish to emulate the native `<select>` keyboard behavior, then the focus should be put onto the select component after the user has made a selection.